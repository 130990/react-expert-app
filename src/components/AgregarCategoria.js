import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AgregarCategoria = ({ setCategorias }) => {
    const [inputValue, setInputValue] = useState('');
    const handleCambiarInput = (e) => {        
        setInputValue(e.target.value)
    }
    const handlEnviar = (e) => {
        e.preventDefault();
        console.log('Llamado Submit......');
        if (inputValue.length > 2) {
            setCategorias(cate => [inputValue,...cate])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handlEnviar}>
            <input type="text" value={inputValue} onChange={handleCambiarInput} />
        </form>
    )
}

AgregarCategoria.propTypes ={
    setCategorias: PropTypes.func.isRequired
}
