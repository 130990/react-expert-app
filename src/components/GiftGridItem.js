import React from 'react'
import '../index.css';

export const GiftGridItem = ({title, url }) => {
    return (
        <div className="card animate__animated animate__fadein">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
