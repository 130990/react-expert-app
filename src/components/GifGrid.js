import React from 'react'
import { useFecthGifs } from '../hooks/useFecthGifs'
import { GiftGridItem } from './GiftGridItem'

export const GifGrid = ({ categoria }) => {

    const { data: imagenes, loading } = useFecthGifs(categoria);


    return (
        <>
            <h3>{categoria}</h3>
            {loading &&<p className='animate__animated animate__flash'>Cargando imagenes...</p> }
            <div className="card-grid" >
                {
                    imagenes.map((img) => (
                        <GiftGridItem key={img.id} {...img} /> //enviando las imagenes con el spread, permite enviar las propiedades como un solo objeto, q puede desestructurase del otro lado
                    ))
                }
            </div>
        </>
    )
}
