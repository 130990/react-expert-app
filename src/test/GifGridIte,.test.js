import { shallow } from 'enzyme';
import {GiftGridItem} from '../components/GiftGridItem'
import React from 'react';
import '@testing-library/jest-dom'

describe('Prueba de validacion de componente', () => {
    test('Debe de validar correcta ejecucion del componente', () => {
        const ResultadoObtenido = shallow(<GiftGridItem title={'prueba'} url={''} />);
        expect(ResultadoObtenido).toMatchSnapshot();
    })
    
    
})
