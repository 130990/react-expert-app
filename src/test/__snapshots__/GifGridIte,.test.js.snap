// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de validacion de componente Debe de validar correcta ejecucion del componente 1`] = `
<div
  className="card animate__animated animate__fadein"
>
  <img
    alt="prueba"
    src=""
  />
  <p>
    prueba
  </p>
</div>
`;
